### DISCUSSION TITLE
First experiences using the multivendor plugin

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#2162] [Q&A] [answered]
This is probably a premature question given that I think work is being done on this at the moment, but I thought I'd log the problem anyway.  

I've been testing the multivendor plugin with the v2beta code base and the plugin with the changes added as suggested by @michaelbromley in my last [message](https://github.com/vendure-ecommerce/vendure/discussions/2156#discussion-5146564).  

It looks really good and I'm really excited about the prospect of using it. I manually created the bobs-parts seller as suggested in the multivendor comments.  I can then log into bob's account on the admin interface and add products (though I did need to update his channel's role to allow products to be created).  

When I then go to a storefront (remix), and buy two items, one from Bob's store and one from another, I am given the option to select the delivery methods for each, so that works well.  But when I then click on bob's shipping method (standard shipping for Bob's parts), an error is thrown:

```
Error: {"message":"Variable \"$shippingMethodId\" of type \"ID!\" used in position expecting type \"[ID!]!\".","locations":[{"line":1,"column":33},{"line":2,"column":44}]}
```
My guess is that the remix storefront assumes only a single shipping method is needed, whereas the v2 version now has an array of shipping methods to handle multiple sellers, but I may be wrong. Again, I'm happy to wait until v2.0 is released, but if there is a simple fix then it would be great to continue to test.  I assume (hope) I just need to update a type description somewhere?

One other thing, when I was looking at implementing a (far more naive) version of a multi-vendor market place in V1, I wanted the facets to automatically propagate a new sellers, rather than require seller to create facets themselves.  From what I recall, I did this by listening for when a new user was created and then manually creating a channel and associating it with the user and the facets.  I can obviously still do this, but it would be great to know what your thoughts/plans were regarding a facet 'superset' that is used by all sellers.

Many thanks again,

Tom

**Comment by michaelbromley** [comment, marked as answer]
Hi Tom,

Thanks for this feedback! The shipping method issue with the Remix storefront is known, and is due to a breaking change in the v2 shop API. This is being tracked here:

- https://github.com/vendure-ecommerce/storefront-remix-starter/issues/40

You can make the change suggested in the issue in your own storefront and it should resolve that.

For the facets thing, the way you handled it sounds like the approach I would suggest anyway. I would build it in as part of the multivendor plugin and indeed I might add that to the official multivendor example when I work on it later.

**Reply by tlodge** [reply]
Thanks again.  That all makes sense and good to know I was on the right track with the facets.

**Comment by dylviz** [comment]
I just want to ask if you could share a brief overview about how you're going about testing the multivendor plugin. I'm also trying to get my feet wet and just want to make sure I'm going about this the right way. I can only see the multivendor plugin in the multivendor branch on the repo. Is this what you're using?

As opposed to setting up a vendure project the normal way (ie `npx @vendure/create my-app`) then adding in the multivendor plugin. I couldn't see how to get it using this method so I figured I had to do the former.

Oops I just saw that you linked to a message that looks to have a better explanation. Will check that out.

**Reply by tlodge** [reply]
Yep - the other link has a better explanation.  I used the npx @vendure/create@next to get the v2 beta version, then used the commit https://github.com/vendure-ecommerce/vendure/commit/9ba67a0d087513bc890da537399f7c0d44247dbb) for the multi-vendor plugin, and made the couple of changes that @michaelbromley suggested in the other link.  

I then added the multivendor plugin to the vendure-config.ts:

```
 MultivendorPlugin.init({
                    platformFeePercent: 10,
                    platformFeeSKU: 'FEE',
        }),
```

and restarted vendure.  It complained about one of the tables not being in the correct format (I'm using postgres) , but gives the ALTER table sql that I needed to run, which I ran and then all worked fine.  

I think (but could be wrong) that the major branch also has multivendor support in it, but I couldn't get that to compile.

Best of luck!

**Reply by dylviz** [reply]
Ahh yes, I saw that too. I ended up just creating a migration using `sudo ts-node migration.ts generate MultivendorChange` and it seemed to include the correct DB modifications so I just ran it using `sudo ts-node migration.ts run` and it seems to be up and running now! See? I knew we make a great team @tlodge! lol thanks for the help :)
